src/gpl/src/initialPlace.cpp

```diff
@@ -41,6 +43,40 @@
 #include "graphics.h"
 
 #include "utl/Logger.h"
+#include <cuda.h>
+#include <cuda_runtime.h>
+#include "cusparse.h"
+#include "cusolverDn.h"
+#include "cusolverSp.h"
+
+#define CUDA_ERROR(value) do { \
+   cudaError_t m_cudaStat = value; \
+   if(m_cudaStat != cudaSuccess){ \
+       fprintf(stderr, "Error %s at line %d in file %s\n", cudaGetErrorString(m_cudaStat), __LINE__, __FILE__); \
+       exit(-1); \
+   } \
+} while(0)
+#define CUSPARSE_ERROR(value) do { \
+    cusparseStatus_t _m_status = value;\
+    if (_m_status != CUSPARSE_STATUS_SUCCESS){\
+        fprintf(stderr, "Error %d at line %d in file %s\n", (int)_m_status, __LINE__, __FILE__);\
+    exit(-5); \
+    } \
+} while(0)
+
+#define CUSOLVER_ERROR(value) do { \
+    cusolverStatus_t _m_status = value; \
+    if (_m_status != CUSOLVER_STATUS_SUCCESS){ \
+        fprintf(stderr, "Error %d at line %d in file %s\n", (int)_m_status, __LINE__, __FILE__);\
```

Comment:
Error messages should be meaningful to end users which this is not.

Reply:
This macro just helps the code to stop and show where the error is. Or the code will continue to run and return a bunch of zeros. I think the better way is to keep the macro and print where the error is. Meantime, I will put more detailed error message into log file. Is this a good solution?